android/src/main/kotlin/com/onegini/mobile/sdk/flutter/handlers/CustomRegistrationHandler.kt

```diff
@@ -0,0 +1,49 @@
+package com.onegini.mobile.sdk.flutter.handlers
+
+import com.onegini.mobile.sdk.flutter.providers.CustomRegistrationAction
+import java.lang.Exception
+
+class CustomRegistrationHandler {
+    fun actionRespondSuccess(
+        idProvider: String?,
+        customRegistrationActions: ArrayList<CustomRegistrationAction>,
+        token: String?) {
+        val action = getCustomRegistrationAction(idProvider, customRegistrationActions)
+
+        when {
+            action != null ->  action.returnSuccess(token)
+            else -> return // TODO throw platform exception no provider found with that id once error rework is merged
+         }
+    }
+
+    fun actionRespondError(
+        idProvider: String?,
+        customRegistrationActions: ArrayList<CustomRegistrationAction>,
+        error: String?) {
+        if (error == null) {
+            // TODO Thow platform exception
+            return
+        }
+
+        val action = getCustomRegistrationAction(idProvider, customRegistrationActions)
+
+        when {
+            action != null ->  action.returnError(Exception(error))
+            else -> return // TODO throw platform exception
+        }
+    }
+
+    private fun getCustomRegistrationAction(idProvider: String?, customRegistrationActions: ArrayList<CustomRegistrationAction>): CustomRegistrationAction? {
+        if (idProvider == null) return null
```

Comment:
```
return idProvider?.let {  
   customRegistrationActions.find { it.getIdProvider() == idProvider } } ?: null
```

Reply:
(1) reworked it with different errors, 
(2) agree with the longer more descriptive names so left it as is
(3) reworked by using find